src/contracts/pods/EigenPodManager.sol

```diff
@@ -215,6 +224,53 @@ contract EigenPodManager is
         ownerToPod[podOwner].withdrawRestakedBeaconChainETH(destination, shares);
     }
 
+    /// @notice Called by proving service to fulfill partial withdrawal proof requests
+    function proofServiceCallback(
+        WithdrawalCallbackInfo[] calldata callbackInfo
+    ) external onlyProofService nonReentrant {
+        require(proofServiceEnabled, "EigenPodManager.proofServiceCallback: offchain partial withdrawal proofs are not enabled");
+        for(uint256 i = 0; i < callbackInfo.length; i++) {
```

Comment:
wait what.

we should be verifying a single proof that proves the pws for many pods against the same block root.